lib/verifier/post_zephir_verifier.rb

```diff
@@ -0,0 +1,195 @@
+        verify_parseable_ndj(path: input_path)
+        input_linecount = gzip_linecount(path: input_path)
+
+        if output_linecount != input_linecount
+          error(
+            message: sprintf(
+              "output line count (%s = %s) != input line count (%s = %s)",
+              output_path,
+              output_linecount,
+              input_path,
+              input_linecount
+            )
+          )
+        end
+      end
+    end
+
+    # Frequency: ALL
+    # Files: CATALOG_PREP/zephir_upd_YYYYMMDD.json.gz and CATALOG_PREP/zephir_upd_YYYYMMDD_delete.txt.gz
+    #   and potentially CATALOG_PREP/zephir_full_YYYYMMDD_vufind.json.gz
+    # Contents:
+    #   json.gz files: ndj with one catalog record per line
+    #   delete.txt.gz: see `#verify_deletes_contents`
+    # Verify:
+    #   readable
+    #   TODO: deletes file is combination of two component files in TMPDIR?
+    def verify_catalog_prep(date: current_date)
+      delete_file = self.class.dated_derivative(location: :CATALOG_PREP, name: "zephir_upd_YYYYMMDD_delete.txt.gz", date: date)
+      verify_file(path: self.class.dated_derivative(location: :CATALOG_PREP, name: "zephir_upd_YYYYMMDD.json.gz", date: date))
+      if verify_file(path: delete_file)
+        verify_deletes_contents(path: delete_file)
+      end
+      if date.last_of_month?
+        verify_file(path: self.class.dated_derivative(location: :CATALOG_PREP, name: "zephir_full_YYYYMMDD_vufind.json.gz", date: date))
+      end
+    end
+
+    # Verify contents of the given file consists of catalog record IDs (9 digits)
+    # or blank lines
+    def verify_deletes_contents(path:)
+      Zlib::GzipReader.open(path).each_line do |line|
+        if line != "\n" && !line.match?(/^\d{9}$/)
+          error message: "Unexpected line in #{path} (was '#{line.strip}'); expecting catalog record ID (9 digits)"
+        end
+      end
+    end
+
+    # Frequency: DAILY
+    # Files: TMPDIR/vufind_incremental_YYYY-MM-DD_dollar_dup.txt.gz
+    # Contents: historically undallarized uc1 HTIDs (e.g., uc1.b312920) one per line
+    # Verify:
+    #  readable
+    #  empty
+    def verify_dollar_dup(date: current_date)
+      dollar_dup = self.class.dated_derivative(location: :TMPDIR, name: "vufind_incremental_YYYY-MM-DD_dollar_dup.txt.gz", date: date)
+      if verify_file(path: dollar_dup)
+        gz_count = gzip_linecount(path: dollar_dup)
+        if gz_count.positive?
+          error message: "spurious dollar_dup lines: #{dollar_dup} should be empty (found #{gz_count} lines)"
+        end
+      end
+    end
+
+    # Frequency: MONTHLY
+    # Files:
+    #   INGEST_BIBRECORDS/groove_full.tsv.gz
+    #   INGEST_BIBRECORDS/zephir_ingested_items.txt.gz
+    # Contents: TODO
+    # Verify: readable
+    def verify_ingest_bibrecords(date: current_date)
+      if date.last_of_month?
+        verify_file(path: self.class.derivative(location: :INGEST_BIBRECORDS, name: "groove_full.tsv.gz"))
```

Comment:
Could add verification for the contents, probably not a top priority.

Reply:
added, implemented and tested Derivative::IngestBibrecord